cmd/xdpcap/program.go

```diff
@@ -161,14 +162,17 @@ func newProgram(filter []bpf.Instruction, action xdpAction, perfMap *ebpf.Map) (
 		asm.Return(),
 	)
 
-	prog, err := ebpf.NewProgram(
-		&ebpf.ProgramSpec{
-			Name:         "xdpcap_filter",
-			Type:         ebpf.XDP,
-			Instructions: insns,
-			License:      "GPL",
-		},
-	)
+	progSpec := &ebpf.ProgramSpec{
+		Name:         "xdpcap_filter",
+		Type:         ebpf.XDP,
+		Instructions: insns,
+		License:      "GPL",
+	}
+	if xdpFrags {
+		progSpec.Flags = progSpec.Flags | unix.BPF_F_XDP_HAS_FRAGS
```

Comment:
Sure we can try that, I'll add a bit of logging so that it's clear to the user what is happening

Reply:
Found a small issue with this. The issue is that we first load the program in legacy or multi-buffer mode and then we hook it up to the XDP program. However, if there is an error to be thrown it will be thrown during the attach phase. So I can remove the flag but the `newProgram` will still need a boolean flag to load the program in multi-buffer mode or not and that flag will be set in the for loop for the actions list.

Reply:
Yes, it still needs to be plumbed through to `newProgram` (that's what I was trying to illustrate with my pseudo code above).

Reply:
fixed